### PR TITLE
PMM-8309: [Pipelines] Migrate pmm-server:dev-latest to ECR

### PR DESCRIPTION
**Pull request by Sudokamikaze**
Builds:
* AWS: https://pmm.cd.percona.com/job/aws-staging-start-illiatests/
* Autobuild: https://pmm.cd.percona.com/job/pmm2-server-autobuild-illiatests/3/
* UI tests: https://pmm.cd.percona.com/job/pmm2-ui-tests-illiatests/2/

### CODE REVIEW

File: pmm/pmm2-api-tests.groovy [hunk 1]
```diff
@@ -76,37 +76,41 @@ pipeline {
...
-                    -e PERCONA_TEST_DBAAS=0 \
-                    -e PERCONA_TEST_SAAS_HOST=check-dev.percona.com:443 \
-                    -e PERCONA_TEST_CHECKS_PUBLIC_KEY=RWTg+ZmCCjt7O8eWeAmTLAqW+1ozUbpRSKSwNTmO+exlS5KEIPYWuYdX \
-                    -p 80:80 \
-                    -p 443:443 \
-                    -v \${PWD}/testdata/checks:/srv/checks \
-                    \${DOCKER_VERSION}
+                withCredentials([aws(accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AMI/OVF', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY')]) {
+                    sh '''
+                        aws ecr-public get-login-password --region us-east-1 | docker login -u AWS --password-stdin public.ecr.aws/e7j3v3n0
```
Comment (BupycHuk) [inline pmm/pmm2-api-tests.groovy:87]: Do we really need to login in this step?

**Review by BupycHuk — COMMENTED** [review]
Will our docker repos be public for our contributors?
If yes probably we don't need to docker login to run docker images in a few places where we do it?

**Comment by Sudokamikaze** [comment]
> Will our docker repos be public for our contributors?
> If yes probably we don't need to docker login to run docker images in a few places where we do it?

It's used to decrease billing, as I remember if logged in, it will be billed as inter-AWS pull and will cost 0$
Without it'll cost 0.02$

**Review by EvgeniyPatlan — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

**Review by nikita-b — APPROVED** [review]

**Review by puneet0191 — APPROVED** [review]

**Review by atymchuk — APPROVED** [review]
Excellent job, thanks Illia!

**Pull request merged**